tabrepo/benchmark/task/openml/task_utils.py

```diff
@@ -1,11 +1,120 @@
 from __future__ import annotations
 
-import time
 
+import boto3
+import logging
+import os
 import openml
 import pandas as pd
+import time
+import xmltodict
+from pathlib import Path
+
 from openml import OpenMLSupervisedTask
 from openml.exceptions import OpenMLServerException
+from tabflow.utils.s3_utils import parse_s3_uri
+
+logger = logging.getLogger(__name__)
+
+
+def download_task_from_s3(task_id: int, s3_dataset_cache: str = None) -> bool:
+    """
+    Downloads the task and dataset from S3 if available.
+    """
+    if s3_dataset_cache is None:
+        return False
```

Comment:
Don't make a function that does nothing when it is called. If you are calling a function when it is useless, then don't call it in the first place.